pkg/routing/table.go

```diff
@@ -1,135 +1,211 @@
 package routing
 
 import (
-	"bytes"
-	"encoding/json"
-	"fmt"
-	"strings"
+	"context"
+	"net/http"
 	"sync"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+
+	httpv1alpha1 "github.com/kedacore/http-add-on/operator/apis/http/v1alpha1"
+	"github.com/kedacore/http-add-on/operator/generated/informers/externalversions"
+	informershttpv1alpha1 "github.com/kedacore/http-add-on/operator/generated/informers/externalversions/http/v1alpha1"
+	listershttpv1alpha1 "github.com/kedacore/http-add-on/operator/generated/listers/http/v1alpha1"
 )
 
-type TableReader interface {
-	Lookup(string) (*Target, error)
-	Hosts() []string
-	HasHost(string) bool
+type Table interface {
+	Start(ctx context.Context) error
+	Route(req *http.Request) *httpv1alpha1.HTTPScaledObject
+	HasSynced() bool
 }
-type Table struct {
-	fmt.Stringer
-	m map[string]Target
-	l *sync.RWMutex
+
+type table struct {
+	// TODO(pedrotorres): remove after upgrading k8s.io/client-go to v0.27.0
```

Comment:
Are you waiting for KEDA using that dep? If yes, [this PR](https://github.com/kedacore/keda/pull/4473) could be interesting for you and will be merged soon

Reply:
Yes! When you PR gets merged and we update the KEDA version here, we can make the modifications I’ve commented on the code.

Reply:
Yes! When you PR gets merged and we update the KEDA version here, we can make the modifications I’ve commented on the code.

Reply:
@JorTurFer, given your PR was merged, should we wait for KEDA 2.10.2 or 2.11.0 before merging this patch?

Reply:
TBH, I wouldn't wait. This project is in beta and we also have [the compatibility table](https://github.com/kedacore/http-add-on/blob/main/docs/install.md#compatibility-table):
![image](https://github.com/kedacore/http-add-on/assets/36899226/3f72f197-1954-430a-add8-c55b9252ee3a)
If this breaks something with other KEDA versions, we can just limit the compatibility

Reply:
Are you suggesting that I update the dependency of this project to the master version of KEDA? Or that I uncomment the code that depends on k8s.io/client-go@v0.27.0 later, on another PR?

Reply:
I'd use main branch from KEDA in order to release the feature, and once we release KEDA v2.11 we can update it again to fix the version

Reply:
I'll do that